src/test/java/com/debatetimer/controller/member/MemberDocumentTest.java

```diff
@@ -0,0 +1,141 @@
+package com.debatetimer.controller.member;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.restdocs.payload.JsonFieldType.ARRAY;
+import static org.springframework.restdocs.payload.JsonFieldType.NUMBER;
+import static org.springframework.restdocs.payload.JsonFieldType.STRING;
+import static org.springframework.restdocs.payload.PayloadDocumentation.fieldWithPath;
+import static org.springframework.restdocs.request.RequestDocumentation.parameterWithName;
+
+import com.debatetimer.controller.BaseDocumentTest;
+import com.debatetimer.controller.RestDocumentationRequest;
+import com.debatetimer.controller.RestDocumentationResponse;
+import com.debatetimer.controller.Tag;
+import com.debatetimer.dto.member.MemberCreateRequest;
+import com.debatetimer.dto.member.MemberCreateResponse;
+import com.debatetimer.dto.member.TableResponse;
+import com.debatetimer.dto.member.TableResponses;
+import com.debatetimer.dto.member.TableType;
+import com.debatetimer.exception.custom.DTClientErrorException;
+import com.debatetimer.exception.errorcode.ClientErrorCode;
+import io.restassured.http.ContentType;
+import java.util.List;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+public class MemberDocumentTest extends BaseDocumentTest {
+
+    @Nested
+    class CreateMember {
+
+        private final RestDocumentationRequest requestDocument = request()
+                .tag(Tag.MEMBER_API)
+                .description("멤버 생성")
+                .requestBodyField(
+                        fieldWithPath("nickname").type(STRING).description("멤버 닉네임")
+                );
+
+        private final RestDocumentationResponse responseDocument = response()
+                .responseBodyField(
+                        fieldWithPath("id").type(NUMBER).description("멤버 ID"),
+                        fieldWithPath("nickname").type(STRING).description("멤버 닉네임")
+                );
+
+        @Test
+        void 회원_생성_성공() {
+            MemberCreateRequest request = new MemberCreateRequest("커찬");
+            MemberCreateResponse response = new MemberCreateResponse(1L, "커찬");
+            when(memberService.createMember(request)).thenReturn(response);
+
+            var document = document("member/create", 201)
+                    .request(requestDocument)
+                    .response(responseDocument)
+                    .build();
+
+            given(document)
+                    .contentType(ContentType.JSON)
+                    .body(request)
+                    .when().post("/api/member")
+                    .then().statusCode(201);
+        }
+
+        @EnumSource(
+                value = ClientErrorCode.class,
```

Comment:
[제안]
이거 문자열로 관리하기가 힘들 것 같은데 MethodSource 사용해보는 건 어떨까요?
관리 용이성과 타입안전성에 좋을 것 같아요.

```suggestion
        @MethodSource("memberCreateError")
        @ParameterizedTest
        void 회원_생성_실패(ClientErrorCode errorCode) {
            MemberCreateRequest request = new MemberCreateRequest("커찬");
            when(memberService.createMember(request)).thenThrow(new DTClientErrorException(errorCode));

            var document = document("member/create", errorCode)
                    .request(requestDocument)
                    .response(ERROR_RESPONSE)
                    .build();

            given(document)
                    .contentType(ContentType.JSON)
                    .body(request)
                    .when().post("/api/member")
                    .then().statusCode(errorCode.getStatus().value());
        }

        private static Stream<ClientErrorCode> memberCreateError() {
            return Stream.of(
                    ClientErrorCode.INVALID_MEMBER_NICKNAME_FORM,
                    ClientErrorCode.INVALID_MEMBER_NICKNAME_LENGTH
            );
        }
```

Reply:
개인적으로 비선호하긴 합니다. 저는 어떤 ERROR CODE를 사용하는지 최대한 테스트와 붙여 놓은 것을 선호하긴 합니다.
물론 컴파일 환경에서 관리할 수 있다는 것이 큰 장점이긴 하지만, 저 에러 코드가 바뀌는 케이스를 거이 못봐서 `@EnumSource`를 이용했습니다.

Reply:
인정합니다. 개인적인 취향으로 제가 MethodSource를 선호하는 이유는 문자열 관리를 싫어하고 + 오타가 많아서 그런가봐요 😓  지금으로도 괜찮다 생각해요!